### ISSUE TITLE
Runtime error: Attempted to compile AOT function without the compiler used by "numpy.distutils"

### ISSUE THREAD

**Issue by Lukikrew** [marcelotduarte/cx_Freeze#1540]
Our setup.py:

```
from cx_Freeze import setup, Executable

build_options = {'packages': ["numba", "numpy", "PySide6", "cv2", "PIL", "pyautogui", "Crypto", "shiboken6"],
                 "zip_exclude_packages": []
}

import sys
base = 'Win32GUI' if sys.platform=='win32' else None



exe = Executable(script = 'app.py', base=base, target_name = 'App', icon = 'icons/icon.icns')

setup(name='App',
      version = '1.6.0',
      description = 'App',
      options = {'build_exe': build_options},
      executables = [exe])
```

We where able to compile the app without any issue.
We sent the app then to multiple users (10 Macs, all running on M1 and v12.4) and about 50% of them receive this error, when they try to open it:

```
Last login: Sat Jun 18 13:25:53 on console
/Applications/App-1.6.0.app/Contents/MacOS/app ; exit;

The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
mitchells-mini:~ mitchellvincent$ /Applications/app-1.6.0.app/Contents/MacOS/app ; exit;
Traceback (most recent call last):
  File "/Users/mainvenv/lib/python3.8/site-packages/cx_Freeze/initscripts/__startup__.py", line 120, in run
  File "/Users/mainvenv/lib/python3.8/site-packages/cx_Freeze/initscripts/Console.py", line 16, in run
  File "app.py", line 6, in <module>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 655, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 618, in _load_backward_compatible
  File "<frozen zipimport>", line 259, in load_module
  File "/Users/main.py", line 48, in <module>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 655, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 618, in _load_backward_compatible
  File "<frozen zipimport>", line 259, in load_module
  File "/Users/image_manager.py", line 7, in <module>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 655, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 618, in _load_backward_compatible
  File "<frozen zipimport>", line 259, in load_module
  File "/Users/image_worker.py", line 7, in <module>
  File "/Users/modules/apply_effects.py", line 5, in <module>
  File "/Users/modules/filters.py", line 5, in <module>
  File "/Users/modules/dithering.py", line 6, in <module>
  File "/Users/mainvenv/lib/python3.8/site-packages/numba/pycc/cc.py", line 65, in __init__
  File "/Users/mainvenv/lib/python3.8/site-packages/numba/pycc/platform.py", line 78, in __init__
  File "/Users/mainvenv/lib/python3.8/site-packages/numba/pycc/platform.py", line 121, in _raise_external_compiler_error
RuntimeError: Attempted to compile AOT function without the compiler used by `numpy.distutils` present. If using conda try:

#> conda install clang_osx-64 clangxx_osx-64
logout

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.
Deleting expired sessions...13 completed.

[Process completed]
```

any idea what can cause this problem?

Tested with python 3.8.1 and 3.9.9, same issue.

cx_Freeze version:
cx-Freeze @ git+https://github.com/marcelotduarte/cx_Freeze.git@7596026a25e64dffb9b9d88d913dd8dce0e6c5fa

Numpy & Numba versions:
numba==0.55.2
numpy==1.22.4

**Comment by marcelotduarte** [comment]
What is the version of cx_Freeze? Why not use recent versions like 6.10, 6.11, 6.11.1?
in conda have binaries for cx_Freeze, that can be installed directly by conda

`conda install -c conda-forge cx_freeze`

**Comment by Lukikrew** [comment]
@marcelotduarte thank you for your answer.

Because as soon as I use any other cx versions, I run into a bunch of issues.

For example the conda you mentioned above results in:


```
Traceback (most recent call last):
  File "/Users/luki/Desktop/app/cx-setup2.py", line 15, in <module>
    setup(name='app',
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/cx_Freeze/__init__.py", line 68, in setup
    setuptools.setup(**attrs)
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/setuptools/__init__.py", line 155, in setup
    return distutils.core.setup(**attrs)
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 148, in setup
    return run_commands(dist)
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 163, in run_commands
    dist.run_commands()
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 967, in run_commands
    self.run_command(cmd)
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 986, in run_command
    cmd_obj.run()
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/cx_Freeze/command/bdist_mac.py", line 407, in run
    self.run_command("build")
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/setuptools/_distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 986, in run_command
    cmd_obj.run()
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/cx_Freeze/command/build.py", line 176, in run
    self.run_command(cmd_name)
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/setuptools/_distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 986, in run_command
    cmd_obj.run()
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/cx_Freeze/command/build_exe.py", line 268, in run
    freezer.Freeze()
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/cx_Freeze/freezer.py", line 652, in Freeze
    self._post_freeze_hook()
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/cx_Freeze/freezer.py", line 863, in _post_freeze_hook
    self.darwin_tracker.finalizeReferences()
  File "/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/cx_Freeze/darwintools.py", line 631, in finalizeReferences
    raise DarwinException(
cx_Freeze.exception.DarwinException: finalizeReferences() failed to resolve path [@rpath/libc++.1.dylib] in file [/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/numba/_dispatcher.cpython-39-darwin.so].
```

**Comment by marcelotduarte** [comment]
in conda list it appears as cx_freeze 6.11.1?

**Comment by Lukikrew** [comment]
@marcelotduarte  yes it does.

We also tried it with a pyenv instead of conda, with the same result. 

`Attempted to compile AOT function without the compiler used by `numpy.distutils` present. If using conda try:`

And on cx_freeze 6.10 and 6.11 I get:

`cx_Freeze.exception.DarwinException: finalizeReferences() failed to resolve path [@rpath/libpyside6.abi3.6.2.dylib] in file [/Users/luki/miniforge3/envs/main39/lib/python3.9/site-packages/PySide6/QtCore.abi3.so]
`

**Comment by marcelotduarte** [comment]
PR #1521 fixes this error on v 6.11.1, so with v6.11.1, this error should not arise.
Try to use zip_include_packages=["PySide6"] to see the results.

On the other hand, the original error, about the use o AOT (https://numba.pydata.org/numba-doc/dev/user/pycc.html) it must use a c-compiler, maybe that's why on some machines it works and on others it doesn't...

**Comment by marcelotduarte** [comment]
[Release 6.15.3](https://github.com/marcelotduarte/cx_Freeze/releases/tag/6.15.3) is out!

Can you test it and give me feedback?
